compiler/aarch64/codegen/ControlFlowEvaluator.cpp

```diff
@@ -480,12 +480,65 @@ OMR::ARM64::TreeEvaluator::tableEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
 	}
 
+TR::Register *
+evaluateNULLCHKWithPossibleResolve(TR::Node *node, bool needsResolve, TR::CodeGenerator *cg)
+   {
+   // NOTE:
+   // If in the future no code is generated for the null check, just evaluate the
+   // child and decrement its use count UNLESS the child is a pass-through node
+   // in which case some kind of explicit test or indirect load must be generated
+   // to force the null check at this point.
+
+   TR::Node *firstChild = node->getFirstChild();
+   TR::ILOpCode &opCode = firstChild->getOpCode();
+   TR::Node *reference = NULL;
+   TR::Compilation *comp = cg->comp();
+
+   reference = node->getNullCheckReference();
+
```

Comment:
You are missing the part that actually does the NULLness check and branches to a snippet for handling.  For an explicit check, right about here you will need code that looks something like this:
```
   TR::LabelSymbol *snippetLabel = generateLabelSymbol(cg);
   cg->addSnippet(new (cg->trHeapMemory()) TR::ARM64HelperCallSnippet(cg, node, snippetLabel, node->getSymbolReference()));

   TR::Register *referenceReg = cg->evaluate(reference);
   TR::InstOpCode::Mnemonic compareOp = comp->useCompressedPointers() ? TR::InstOpCode::cbzw : TR::InstOpCode::cbzx;
   TR::Instruction *cbzInstruction = generateCompareBranchInstruction(cg, compareOp, node, divisorReg, snippetLabel, NULL);

   // consider all registers live
   cbzInstruction->setNeedsGCMap(0xffffffff);
```

This code will have to reconcile with the logic that you implemented below (especially the node evaluation).

Also, have a look at what Power and X does for compressed refs when the node is an `l2a` and how you have to reach for the actual reference.  You'll probably need something like that as well.